File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/multirt/Runtime.java
package net.kdt.pojavlaunch.multirt;

import java.util.Objects;

/**
 * Javaランタイムの情報を保持するクラス
 */
public class Runtime {
    public final String name;
    public final String versionString;
    public final String arch;
    public final int javaVersion;
    public boolean isProvidedByLauncher = false;

    /**
     * 破損または未知のランタイム用のコンストラクタ
     * @param name ランタイム名
     */
    public Runtime(String name) {
        this.name = name;
        this.versionString = null;
        this.arch = null;
        this.javaVersion = 0;
    }

    /**
     * 完全な情報を持つランタイム用のコンストラクタ
     * @param name ランタイム名
     * @param versionString バージョン文字列
     * @param arch アーキテクチャ
     * @param javaVersion Javaメジャーバージョン
     */
    Runtime(String name, String versionString, String arch, int javaVersion) {
        this.name = name;
        this.versionString = versionString;
        this.arch = arch;
        this.javaVersion = javaVersion;
    }
    
    /**
     * 名前が同じであれば同じランタイムと見なします。
     */
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Runtime runtime = (Runtime) o;
        return name.equals(runtime.name);
    }

    /**
     * 名前からハッシュコードを生成します。
     */
    @Override
    public int hashCode() {
        return Objects.hash(name);
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/utils/runtime/Runtime.kt
package com.arata.yukarilauncher.utils.runtime

import java.util.Objects

/** Javaランタイム（JRE）の情報を保持するデータクラス。 */
class Runtime(
    /** ランタイムの名前（ディレクトリ名）。 */
    val name: String,
    /** バージョン文字列（releaseファイルから取得）。 */
    val versionString: String?,
    /** アーキテクチャ文字列（例: aarch64）。 */
    val arch: String?,
    /** Javaメジャーバージョン番号。 */
    val javaVersion: Int
) {
    /** ランチャーに組み込みで提供されているランタイムかどうか。 */
    var isProvidedByLauncher: Boolean = false

    /** バージョン情報がないランタイム用のコンストラクタ。 @param name ランタイム名 */
    constructor(name: String) : this(name, null, null, 0)

    /**
     * 等価比較（名前ベース）。
     * @param other 比較対象
     * @return 等しい場合は true
     */
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val runtime = other as Runtime
        return name == runtime.name
    }

    /** ハッシュコードを返す（名前ベース）。 @return ハッシュコード */
    override fun hashCode(): Int = Objects.hash(name)
}
